error: ‘TLSDataAccumulator’ does not name a type

gcc 9.3 with c++11
opencv 4.5 with latest git commit
[ 82%] Building CXX object modules/python2/CMakeFiles/opencv_python2.dir/__/src2/cv2.cpp.o
cd /data/software/opencv/build/modules/python2 && /usr/bin/ccache /usr/bin/c++  -DCERES_FOUND -DCV_COLLECT_IMPL_DATA -DVK_NO_PROTOTYPES -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dopencv_python2_EXPORTS -I/data/software/opencv/build/3rdparty/ippicv/ippicv_lnx/icv/include -I/data/software/opencv/build/3rdparty/ippicv/ippicv_lnx/iw/include -I/data/software/opencv/build -I/data/software/opencv/3rdparty/include -I/data/software/opencv/modules/python/python2/include -I/data/software/opencv/build/modules/python2 -I/data/software/opencv_contrib/modules/cudev/include -I/data/software/opencv/modules/core/include -I/data/software/opencv_contrib/modules/cudaarithm/include -I/data/software/opencv/modules/flann/include -I/data/software/opencv_contrib/modules/hdf/include -I/data/software/opencv/modules/imgproc/include -I/data/software/opencv_contrib/modules/intensity_transform/include -I/data/software/opencv/modules/ml/include -I/data/software/opencv_contrib/modules/phase_unwrapping/include -I/data/software/opencv_contrib/modules/plot/include -I/data/software/opencv_contrib/modules/quality/include -I/data/software/opencv_contrib/modules/reg/include -I/data/software/opencv_contrib/modules/surface_matching/include -I/data/software/opencv_contrib/modules/viz/include -I/data/software/opencv_contrib/modules/alphamat/include -I/data/software/opencv_contrib/modules/cudafilters/include -I/data/software/opencv_contrib/modules/cudaimgproc/include -I/data/software/opencv_contrib/modules/cudawarping/include -I/data/software/opencv/modules/dnn/include -I/data/software/opencv_contrib/modules/dnn_superres/include -I/data/software/opencv/modules/features2d/include -I/data/software/opencv_contrib/modules/freetype/include -I/data/software/opencv_contrib/modules/fuzzy/include -I/data/software/opencv_contrib/modules/hfs/include -I/data/software/opencv_contrib/modules/img_hash/include -I/data/software/opencv/modules/imgcodecs/include -I/data/software/opencv_contrib/modules/line_descriptor/include -I/data/software/opencv/modules/photo/include -I/data/software/opencv_contrib/modules/saliency/include -I/data/software/opencv_contrib/modules/text/include -I/data/software/opencv/modules/videoio/include -I/data/software/opencv_contrib/modules/xphoto/include -I/data/software/opencv/modules/calib3d/include -I/data/software/opencv_contrib/modules/cnn_3dobj/include -I/data/software/opencv_contrib/modules/cudacodec/include -I/data/software/opencv_contrib/modules/cudafeatures2d/include -I/data/software/opencv_contrib/modules/cudastereo/include -I/data/software/opencv_contrib/modules/cvv/include -I/data/software/opencv_contrib/modules/datasets/include -I/data/software/opencv/modules/highgui/include -I/data/software/opencv_contrib/modules/mcc/include -I/data/software/opencv/modules/objdetect/include -I/data/software/opencv_contrib/modules/ovis/include -I/data/software/opencv_contrib/modules/rapid/include -I/data/software/opencv_contrib/modules/rgbd/include -I/data/software/opencv_contrib/modules/shape/include -I/data/software/opencv_contrib/modules/structured_light/include -I/data/software/opencv/modules/video/include -I/data/software/opencv_contrib/modules/xfeatures2d/include -I/data/software/opencv_contrib/modules/ximgproc/include -I/data/software/opencv_contrib/modules/xobjdetect/include -I/data/software/opencv_contrib/modules/aruco/include -I/data/software/opencv_contrib/modules/bgsegm/include -I/data/software/opencv_contrib/modules/bioinspired/include -I/data/software/opencv_contrib/modules/ccalib/include -I/data/software/opencv_contrib/modules/cudabgsegm/include -I/data/software/opencv_contrib/modules/cudalegacy/include -I/data/software/opencv_contrib/modules/cudaobjdetect/include -I/data/software/opencv_contrib/modules/dpm/include -I/data/software/opencv_contrib/modules/face/include -I/data/software/opencv/modules/gapi/include -I/data/software/opencv_contrib/modules/optflow/include -I/data/software/opencv_contrib/modules/sfm/include -I/data/software/opencv/modules/stitching/include -I/data/software/opencv_contrib/modules/tracking/include -I/data/software/opencv_contrib/modules/cudaoptflow/include -I/data/software/opencv_contrib/modules/superres/include -I/data/software/opencv_contrib/modules/videostab/include -I/data/software/opencv/modules/python/src2 -I/data/software/opencv/build/modules/python_bindings_generator -isystem /usr/include/gdal -isystem /usr/include/coin -isystem /data/bin/intel/oneapi/mkl/latest/include -isystem /usr/include/python2.7 -isystem /usr/local/lib/python2.7/dist-packages/numpy/core/include  -fsigned-char -ffast-math -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread  -pg -g --coverage  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp -Wno-unused-function -Wno-deprecated-declarations -Wno-overloaded-virtual -Wno-undef -O3 -DNDEBUG  -DNDEBUG -fPIC   -std=c++11 -o CMakeFiles/opencv_python2.dir/__/src2/cv2.cpp.o -c /data/software/opencv/modules/python/src2/cv2.cpp
In file included from /data/software/opencv/modules/highgui/include/opencv2/highgui/highgui_c.h:46,
                 from /data/software/opencv_contrib/modules/cnn_3dobj/include/opencv2/cnn_3dobj.hpp:69,
                 from /data/software/opencv/build/modules/python_bindings_generator/pyopencv_generated_include.h:88,
                 from /data/software/opencv/modules/python/src2/cv2.cpp:39:
/data/software/opencv/modules/imgproc/include/opencv2/imgproc/imgproc_c.h:952: warning: "CV_FILLED" redefined
  952 | #define CV_FILLED -1
      |
In file included from /usr/local/include/caffe/blob.hpp:8,
                 from /data/software/opencv_contrib/modules/cnn_3dobj/include/opencv2/cnn_3dobj.hpp:61,
                 from /data/software/opencv/build/modules/python_bindings_generator/pyopencv_generated_include.h:88,
                 from /data/software/opencv/modules/python/src2/cv2.cpp:39:
/usr/local/include/caffe/common.hpp:78: note: this is the location of the previous definition
   78 | #define CV_FILLED cv::FILLED
      |
In file included from /data/software/opencv/modules/core/include/opencv2/core/utility.hpp:1205,
                 from /data/software/opencv/modules/core/include/opencv2/core/utils/tls.hpp:8,
                 from /data/software/opencv/modules/python/src2/cv2.cpp:37:
/data/software/opencv/modules/core/include/opencv2/core/utils/instrumentation.hpp:88:9: error: ‘TLSDataAccumulator’ in namespace ‘cv’ does not name a template type
   88 |     cv::TLSDataAccumulator<NodeDataTls> m_tls;
      |         ^~~~~~~~~~~~~~~~~~
make[2]: *** [modules/python2/CMakeFiles/opencv_python2.dir/build.make:66: modules/python2/CMakeFiles/opencv_python2.dir/__/src2/cv2.cpp.o] Error 1
make[2]: Leaving directory '/data/software/opencv/build'
make[1]: *** [CMakeFiles/Makefile2:22551: modules/python2/CMakeFiles/opencv_python2.dir/all] Error 2
make[1]: Leaving directory '/data/software/opencv/build'
make: *** [Makefile:166: all] Error 2

--- a/modules/core/include/opencv2/core/utils/tls.hpp
+++ b/modules/core/include/opencv2/core/utils/tls.hpp
@@ -5,7 +5,9 @@
 #ifndef OPENCV_UTILS_TLS_HPP
 #define OPENCV_UTILS_TLS_HPP
 
-#include <opencv2/core/utility.hpp>
+#ifndef OPENCV_CORE_UTILITY_H
+#error "tls.hpp must be included after opencv2/core/utility.hpp or opencv2/core.hpp"
+#endif
 
 namespace cv {
 

--- a/modules/python/src2/cv2.cpp
+++ b/modules/python/src2/cv2.cpp
@@ -31,13 +31,14 @@
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/ndarrayobject.h>
 
+#include "opencv2/opencv_modules.hpp"
+#include "opencv2/core.hpp"
 #include "opencv2/core/utils/configuration.private.hpp"
 #include "opencv2/core/utils/logger.hpp"
 #include "opencv2/core/utils/tls.hpp"
 
 #include "pyopencv_generated_include.h"
 #include "opencv2/core/types_c.h"
-#include "opencv2/opencv_modules.hpp"
 #include "pycompat.hpp"
 #include <map>
 

